docs/contributing/coding-guidelines.md

```diff
@@ -18,82 +15,76 @@ $ ng lint
+- Use the store for a feature unless there is a compelling reason not to. We want to keep it consistent throughout the app.
 - Use one common store for the whole app.
-- Using the store doesn’t mean that we need to cache everything.  Caching should be used with intent and where it makes sense.  In general, cms data is a good candidate for caching; application data is not.
+
+**Note**: Using the store does not mean that we need to cache everything. Caching should be used with intent, and where it makes sense. In general, CMS data is a good candidate for caching, while application data is not.
 
 ### Site Context
 
-Site context can be changed for each page. The response data may be different for different site context. Keep this in mind when working on pages.
+Site context can be changed for each page. The response data may be different for different site contexts. Keep this in mind when working on pages.
 
 Also, logged-in users and anonymous users may see different response data. When working on pages, take into consideration that a user can change their login status through login or logout.
 
 ### Naming Conventions
 
-Component selectors should always start with "cx-". (Currently, many component selectors start with "y-" but this is being chnaged.)
-
-### Code Structure
+Component selectors should always start with "cx-".
 
-The code structure of each module should be consistent. For example:
+### Modules
 
-- Store-related codes are in the `store` folder
-- Component codes are inside the `component` folder
-- All OCC (API) services are inside `OccModule`
+Try to keep modules as small as possible. In most cases, one module has only one component. Also, we should always try to reduce module dependencies.
 
-However, common UI components are inside the `UI/components` folder.
+The code structure of each module should be consistent. For example, most modules have the following folders:
 
-### Keep Modules Small
+- a `store` folder that contains store-related code
+- a `facade` folder that contains all facade code for that particular module
+- an `occ` folder that contains module-specific services that call the OCC APIs to communicate with the back end
 
-Try to keep modules as small as possible. In most cases, one module has only one component.
+The one exception is for UI components that are common to all modules, which are in `storefrontlib/src/lib/ui/components`.
 
-### Unit Tests, Always
+### Testing
 
 New code must always be covered by unit tests.
 
-### End-To-End Tests
-
-New, UI-oriented features must always be covered by basic UI end-to-end tests. The file names for the tests should end with `e2e-spec.ts`, and for page objects, the file names should end with `po.ts`. 
+With regards to end-to-end tests, new, UI-oriented features must always be covered by basic UI end-to-end tests. The file names for the tests should end with `e2e-spec.ts`, and for page objects, the file names should end with `po.ts`.
```

Comment:
since we move to cypress, this section should be reviewed. 
Page objects (`po.ts`) are not recommended by cypress and are probably not coming back soon to the new e2e setup we have.

Reply:
I would write:
The file names for the tests should end with `.e2e-spec.ts`.
Common functions that can be re-used should be extracted in different files, with the file names ending with `.ts` and should be located in the sub-directory named 'helpers'.

Reply:
Updated. Thank you! 😄